python/lsst/analysis/tools/actions/vector/calcRhoStatistics.py

```diff
@@ -0,0 +1,649 @@
+        optional=True,
+        allowed={
+            method: method
+            for method in [
+                "shot",
+                "jackknife",
+                "sample",
+                "bootstrap",
+                "marked_bootstrap",
+            ]
+        },
+    )
+
+    num_bootstrap = Field[int](
+        doc=("How many bootstrap samples to use for the 'bootstrap' and " "'marked_bootstrap' var methods."),
+        default=500,
+        optional=True,
+    )
+
+    num_threads = Field[int](
+        doc=(
+            "How many OpenMP threads to use during the calculation. "
+            "If set to None, use all the available CPU cores."
+        ),
+        default=None,
+        optional=True,
+    )
+
+    def validate(self):
+        # Docs inherited from base class
+        super().validate()
+        req_params = (self.nbins, self.bin_size, self.min_sep, self.max_sep)
+        num_req_params = sum(param is not None for param in req_params)
+        if num_req_params != 3:
+            msg = (
+                "You must specify exactly three of ``nbins``, ``bin_size``, ``min_sep`` and ``max_sep``"
+                f" in treecorr_config. {num_req_params} parameters were set instead."
+            )
+            raise FieldValidationError(self.__class__.bin_size, self, msg)
+
+        if self.min_sep is not None and self.max_sep is not None:
+            if self.min_sep > self.max_sep:
+                raise FieldValidationError(self.__class__.min_sep, self, "min_sep must be <= max_sep")
+
+        if self.min_rpar is not None and self.max_rpar is not None:
+            if self.min_rpar > self.max_rpar:
+                raise FieldValidationError(self.__class__.min_rpar, self, "min_rpar must be <= max_rpar")
+
+    def toDict(self):
+        # Docs inherited from base class
+        # We are excluding items that are None due to treecorr limitations.
+        # TODO: DM-38480. This override can be removed after treecorr v4.3
+        # makes its way onto rubin-env.
+        return {k: v for k, v in super().toDict().items() if v is not None}
+
+
+class CalcRhoStatistics(KeyedDataAction):
+    r"""Calculate rho statistics
+
+    Rho statistics refer to a collection of correlation functions involving
+    PSF ellipticity and size residuals. They quantify the contribution from PSF
+    leakage due to errors in PSF modeling to the weak lensing shear correlation
+    functions. The standard rho statistics are indexed from 1 to 5, and
+    this action calculates a sixth rho statistic, indexed 0.
+
+    Notes
+    -----
+    The exact definitions of rho statistics as defined in [1]_ are given below.
+    In addition to these five, we also compute the auto-correlation function of
+    the fractional size residuals and call it as the :math:`\rho_0( \theta )`.
+
+    .. math::
+
+       \rho_0(\theta) &= \left\langle \frac{\delta T_{PSF}}{T_{PSF}}(x) \frac{\delta T_{PSF}}{T_{PSF}}(x+\theta) \right\rangle  # noqa: E501, W505
```

Comment:
rho0 is traditionally what people call <e*_psf e_psf>.  So I think using this name for a different statistic is likely to be confusing.

Reply:
rho6 then?

Reply:
Is there any publication where rho0 is defined?

Reply:
I don't know if I've seen it in anything published.  But the SLAC people who do PSF work use that notation pretty regularly.

Reply:
Marco used it here: https://arxiv.org/pdf/2011.03408.pdf  Figure 8.

Reply:
As a quantity not involving any PSF residuals, I'm unhappy with the use of `rho0` to mean ellipticity correlations. I could go with `rho6` although I suspect that'd prove to be a bad choice as well. I could name it something like `mu0` but that go lead this plot becoming undiscoverable :-|

Reply:
Maybe just call it dT_auto or something more direct like that, rather than some code letter/number?  Also, what's the intended use case for it?  The mean <dT> (one-point function) is relevant for WL systematics, but I don't think this combination is directly applicable to WL.  Does LSS care about this?

Reply:
The auto-correlation function of dT/T is what was called as `rho3` in [Melchior et al. (2015)](https://ui.adsabs.harvard.edu/abs/2015MNRAS.449.2219M/abstract), which you denoted as `rho'3` statistic in your 2016 paper. The reason why it is interesting here, not necessarily from a scientific standpoint, but from a pipelines standpoint is that, for reasons not clear to us, we see a significant offset in PSF size on coadds. This serves as a good diagnostic tool to see how our planned changes to PSF coaddition improves the situation. This complements our other tools involving scatter plots and other one-point calculations from it.

Perhaps, I'll just follow your notation call it `rho3prime` or something.

Reply:
Ah, yes.  Peter and I disagreed at first about how dT errors propagate to xi+.  Originally he thought his `rho3 * <|e|^2>` was the right systematic error on xi+.  But I convinced him that the e terms couldn't be brought out of the angle brackets.  They needed to stay as what I ended up calling rho3.  `<e dT/T, e dT/T>`.